### ISSUE TITLE
Using templates to run kubectl commands with `--kubeconfig` for specific SAs not working

### ISSUE THREAD

**Issue by ffilippopoulos** [utilitywarehouse/kube-applier#48]
`STRICT_APPLY` is based on the fact that it will search for a `kube-applier` user under the respective namespace that it wants to apply manifests and craft a kubeConfig based on the templates: https://github.com/utilitywarehouse/kube-applier/tree/master/templates

That behaviour looks broken, since we see that it sticks to the service account that runs `kube-applier` (inCluster config), even though that: https://github.com/kubernetes/kubernetes/issues/43662#issuecomment-294534615 suggests that in principal the approach is correct.

Example output:
```
Error from server (Forbidden): error when creating "/kubernetes-manifests/repo/dev-aws/partner-platform-api/02-network-policies.yaml": networkpolicies.networking.k8s.io is forbidden: User "system:serviceaccount:partner:kube-applier" cannot create resource "networkpolicies" in API group "networking.k8s.io" in the namespace "partner-platform-api"
Error from server (Forbidden): error when creating "/kubernetes-manifests/repo/dev-aws/partner-platform-api/secrets/docker-registry-key.yaml": secrets is forbidden: User "system:serviceaccount:partner:kube-applier" cannot create resource "secrets" in API group "" in the namespace "partner-platform-api"
exit status 1
```

Instead of crafting templates we can directly use --token flag:
```
kubectl get secret --token=xxxxx
```
grabbing the token from the respective kube-applier SA secret that looks to work correctly.
Also that will save us from creating and deleting temp files. 
We will need to exclude the token from the output as it hold a valid users credentials but still show what we are applying.

**Comment by ffilippopoulos** [comment]
should be fixed by: https://github.com/utilitywarehouse/kube-applier/pull/51 and https://github.com/utilitywarehouse/kube-applier/pull/52.

Will confirm https://github.com/utilitywarehouse/kube-applier/releases/tag/2.2.7 solves it and close

**Comment by ffilippopoulos** [comment]
fixed in v2.2.7
